app/helpers/settings_helper.rb

```diff
@@ -2,42 +2,63 @@ module SettingsHelper
   def field(form, key, opts = {})
     html = '<div class="form-group">'
 
-    case form.object.column_type(key)
+    field_resolver(form.object.column_type(key), html, form, key, opts)
+
+    html << "</div>"
+    html.html_safe
+  end
+
+  private
+  def field_resolver(type, html, form, key, opts)
+    case type
     when :hidden
       return form.hidden_field(key)
```

Comment:
Does this code still work? It seems that `field_resolver` should change `html` destructively.